cart/domain/cart/discount.go

```diff
@@ -0,0 +1,165 @@
+package cart
+
+import (
+	"flamingo.me/flamingo-commerce/v3/price/domain"
+)
+
+type (
+	// AppliedDiscount value object - generic reference for a discount
+	AppliedDiscount struct {
+		Code    string       // unique code of discount
+		Title   string       // readable name of discount
+		Applied domain.Price // how much of the discount has been subtracted from cart price
+		Type    string       // to distinguish between discounts
+	}
```

Comment:
Code - fachlicher schlüssel der Kampagne (CampaignCode)
Type - ggf zusammen mit Code der Schlüssel 
--  Const Type?  Coupon 
CouponCode - wenn Type Coupon
Applied - Document that negative price should be used
Title -> Rename to Label

IsItemRelated bool


=> Dann ItemDiscount Value object löschen

Reply:
Remove error 
-> In case add returns error, break and log